public/app/features/correlations/Forms/TransformationsEditor.tsx

```diff
@@ -0,0 +1,318 @@
+                            <IconButton
+                              type="button"
+                              tooltip="Remove transformation"
+                              name={'trash-alt'}
+                              onClick={() => {
+                                remove(index);
+                                const keptValsCopy: Array<{ expression?: string; mapValue?: string } | undefined> = [
+                                  ...keptVals,
+                                ];
+                                keptValsCopy[index] = undefined;
+                                setKeptVals(compact(keptValsCopy));
+                              }}
+                              ariaLabel="Remove transformation"
+                            >
+                              Remove
+                            </IconButton>
+                          </div>
+                        )}
+                      </Stack>
+                    );
+                  })}
+                </div>
+              )}
+              {!readOnly && (
+                <Button
+                  icon="plus"
+                  onClick={() => append({ type: undefined }, { shouldFocus: false })}
+                  variant="secondary"
+                  type="button"
+                  aria-label="add transformation"
+                >
+                  Add transformation
+                </Button>
+              )}
+            </Stack>
+          </>
+        )}
+      </FieldArray>
+    </>
+  );
+};
+
+interface SupportedTransformationTypeDetails {
+  label: string;
+  value: string;
+  description?: string;
+  showExpression: boolean;
+  showMapValue: boolean;
+  requireExpression?: boolean;
+}
+
+function getSupportedTransTypeDetails(transType: SupportedTransformationType): SupportedTransformationTypeDetails {
+  switch (transType) {
+    case SupportedTransformationType.Logfmt:
+      return {
+        label: 'Logfmt',
+        value: SupportedTransformationType.Logfmt,
+        description: 'Parse provided field with logfmt to get variables',
+        showExpression: false,
+        showMapValue: false,
+      };
+    case SupportedTransformationType.Regex:
+      return {
+        label: 'Regular expression',
+        value: SupportedTransformationType.Regex,
+        description:
+          'Field will be parsed with regex. Use named capture groups to return multiple variables, or a single unnamed capture group to add variable to named map value.',
+        showExpression: true,
+        showMapValue: true,
+        requireExpression: true,
+      };
+    default:
+      return { label: transType, value: transType, showExpression: false, showMapValue: false };
+  }
+}
+
+const getTransformOptions = () => {
+  return Object.keys(SupportedTransformationType).map((key) => {
+    const transType = getSupportedTransTypeDetails(
+      SupportedTransformationType[key as keyof typeof SupportedTransformationType]
+    );
```

Comment:
i think you can do this instead so you don't have to do any ugly type assertion:

```suggestion
  return Object.values(SupportedTransformationType).map((key) => {
    const transType = getSupportedTransTypeDetails(key);
```

Reply:
or probably better:

```suggestion
  return Object.values(SupportedTransformationType).map((transformationType) => {
    const transType = getSupportedTransTypeDetails(transformationType);
```

Reply:
@Elfo404 good call, this worked great!